### PR TITLE
Show filter condition

### PR DESCRIPTION
**Pull request by apstndb**
Currently, Seek Condition/Residual Condition predicates of FilterScan operator are shown but Condition predicate of Filter operator is not shown.
Showing them helps to understand query executions.

This PR fix to show Filter's Condition.

Note: It adds testdata directory which contains QueryPlan as JSON.

## Before
```
> EXPLAIN SELECT s.LastName FROM (SELECT s.LastName FROM Singers AS s WHERE s.FirstName LIKE 'A%' LIMIT 3) s WHERE s.LastName LIKE 'Rich%';
+----+-----------------------------------------------------------------+
| ID | Query_Execution_Plan (EXPERIMENTAL)                             |
+----+-----------------------------------------------------------------+
|  0 | Serialize Result                                                |
|  1 | +- Filter                                                       |
|  2 |    +- Global Limit                                              |
| *3 |       +- Distributed Union                                      |
|  4 |          +- Local Limit                                         |
|  5 |             +- Local Distributed Union                          |
| *6 |                +- FilterScan                                    |
|  7 |                   +- Index Scan (Index: SingersByFirstLastName) |
+----+-----------------------------------------------------------------+
Predicates(identified by ID):
 3: Split Range: STARTS_WITH($FirstName, 'A')
 6: Seek Condition: STARTS_WITH($FirstName, 'A')

1 rows in set (0.02 sec)

```
## After
```
> EXPLAIN SELECT s.LastName FROM (SELECT s.LastName FROM Singers AS s WHERE s.FirstName LIKE 'A%' LIMIT 3) s WHERE s.LastName LIKE 'Rich%';
+----+-----------------------------------------------------------------+
| ID | Query_Execution_Plan (EXPERIMENTAL)                             |
+----+-----------------------------------------------------------------+
|  0 | Serialize Result                                                |
| *1 | +- Filter                                                       |
|  2 |    +- Global Limit                                              |
| *3 |       +- Distributed Union                                      |
|  4 |          +- Local Limit                                         |
|  5 |             +- Local Distributed Union                          |
| *6 |                +- FilterScan                                    |
|  7 |                   +- Index Scan (Index: SingersByFirstLastName) |
+----+-----------------------------------------------------------------+
Predicates(identified by ID):
 1: Condition: STARTS_WITH($LastName, 'Rich')
 3: Split Range: STARTS_WITH($FirstName, 'A')
 6: Seek Condition: STARTS_WITH($FirstName, 'A')

1 rows in set (0.01 sec)
```

### CODE REVIEW

File: query_plan_test.go [hunk 1]
```diff
@@ -18,6 +20,77 @@ func mustNewStruct(m map[string]interface{}) *structpb.Struct {
...
+				{
+					ID:           7,
+					Text:         "                  +- Index Scan (Index: SingersByFirstLastName)",
+				},
+			}},
+	} {
+		t.Run(test.title, func(t *testing.T) {
+			b, err := ioutil.ReadFile(test.file)
+			if err != nil {
+				t.Error(err)
```
Comment (yfuruyama) [inline query_plan_test.go:75]: This should be `t.Fatal` as it can't continue the test.

File: query_plan_test.go [hunk 2]
```diff
@@ -18,6 +20,77 @@ func mustNewStruct(m map[string]interface{}) *structpb.Struct {
...
+			}
+			var plan pb.QueryPlan
+			err = protojson.Unmarshal(b, &plan)
+			if err != nil {
+				t.Error(err)
```
Comment (yfuruyama) [inline query_plan_test.go:80]: Same to the above.

File: query_plan.go [hunk 3]
```diff
@@ -143,7 +151,9 @@ func (n *Node) RenderTreeWithStats(planNodes []*pb.PlanNode) ([]QueryPlanRow, er
 		var predicates []string
 		for _, cl := range planNodes[planNode.ID].GetChildLinks() {
 			child := planNodes[cl.ChildIndex]
-			if child.DisplayName != "Function" || !(cl.GetType() == "Residual Condition" || cl.GetType() == "Seek Condition" || cl.GetType() == "Split Range") {
+			// Known predicates are Condition(Filter) or Seek Condition/Residual Condition(FilterScan) or Split Range(Distributed Union).
+			// Agg is a Function but not a predicate.
+			if child.DisplayName != "Function" || (!strings.HasSuffix(cl.GetType(), "Condition") && cl.GetType() != "Split Range") {
```
Comment (yfuruyama) [inline query_plan.go:156]: By the way, what is the reason to have condition `child.DisplayName != "Function"`? If I understand correctly, `Function` operator is not visible in our query plan.

File: testdata/plans/filter.input.json [hunk 4]
```diff
@@ -0,0 +1,277 @@
+{
```
Comment (yfuruyama) [inline testdata/plans/filter.input.json:1]: Looks good!

File: query_plan.go (hunk 3 above)
Comment (apstndb) [reply query_plan.go:156]: Predicates are subset of `Function` operators which returns bool so `Function` is not visible as a node of the operator tree but needs to be visible as a predicate.
Although, there are no return type information in `PlanNode` so it is whitelisted by the type name of `ChildLink` which link to the `Function` operator.

Currently, I don't yet see a child link like which type is `*Condition` and child is not `Function` so only `ChildLink.type` is needed information to distinguish predicates but I think that to be `Function` is a necessary condition to be a predicate so I require it explicitly.

Comment (apstndb) [reply query_plan.go]: Extract isPredicate function in https://github.com/cloudspannerecosystem/spanner-cli/pull/87/commits/876f8382a762fb6d026afabc8e4f02b9db854ba3

File: query_plan_test.go (hunk 1 above)
Comment (apstndb) [reply query_plan_test.go]: Fixed in https://github.com/cloudspannerecosystem/spanner-cli/pull/87/commits/a05abe82a39ef3bfa3bf9b9100172c1a42d8cc22

File: query_plan_test.go (hunk 2 above)
Comment (apstndb) [reply query_plan_test.go]: Fixed in https://github.com/cloudspannerecosystem/spanner-cli/pull/87/commits/a05abe82a39ef3bfa3bf9b9100172c1a42d8cc22

File: query_plan.go (hunk 3 above)
Comment (yfuruyama) [reply query_plan.go]: Ah that makes sense. Thank you for clarifying it!

**Review by yfuruyama — APPROVED** [review]
LGTM!

**Pull request merged**